src/ShadyCSS/ShadyCSS.js

```diff
@@ -0,0 +1,290 @@
+    let info = {
+      is: elementName,
+      extends: typeExtension,
+      __cssBuild: cssBuild
+    };
+    if (!this.nativeShadow) {
+      StyleTransformer.dom(template.content, elementName);
+    }
+    let ast = parse(cssText);
+    if (this.nativeCss && !this.nativeCssApply) {
+      ApplyShim.transformRules(ast, elementName);
+    }
+    template._styleAst = ast;
+
+    let ownPropertyNames = [];
+    if (!this.nativeCss) {
+      ownPropertyNames = StyleProperties.decorateStyles(template._styleAst, info);
+    }
+    if (!ownPropertyNames.length || this.nativeCss) {
+      let root = this.nativeShadow ? template.content : null;
+      let placeholder = placeholderMap[elementName];
+      let style = this._generateStaticStyle(info, template._styleAst, root, placeholder);
+      template._style = style;
+    }
+    template._ownPropertyNames = ownPropertyNames;
+  },
+  _generateStaticStyle(info, rules, shadowroot, placeholder) {
+    let cssText = StyleTransformer.elementStyles(info, rules);
+    return StyleUtil.applyCss(cssText, info.is, shadowroot, placeholder);
+  },
+  _prepareHost(host) {
+    let is = host.getAttribute('is') || host.localName;
+    let typeExtension;
+    if (is !== host.localName) {
+      typeExtension = host.localName;
+    }
+    let placeholder = placeholderMap[is];
+    let template = templateMap[is];
+    let ast;
+    let ownStylePropertyNames;
+    let cssBuild;
+    if (template) {
+      ast = template._styleAst;
+      ownStylePropertyNames = template._ownPropertyNames;
+      cssBuild = template._cssBuild;
+    }
+    return StyleInfo.set(host,
+      new StyleInfo(
+        ast,
+        placeholder,
+        ownStylePropertyNames,
+        is,
+        typeExtension,
+        cssBuild
+      )
+    );
+  },
+  applyStyle(host, overrideProps) {
+    if (window.CustomStyle) {
+      let CS = window.CustomStyle;
+      if (CS._documentDirty) {
+        CS.findStyles();
+        if (!this.nativeCss) {
+          this._updateProperties(this._documentOwner, this._documentOwnerStyleInfo);
+        }
+        CS.applyStyles();
+        CS._documentDirty = false;
+      }
+    }
+    let styleInfo = StyleInfo.get(host);
+    if (!styleInfo) {
+      styleInfo = this._prepareHost(host);
+    }
+    let is = host.getAttribute('is') || host.localName;
+    Object.assign(styleInfo.overrideStyleProperties, overrideProps);
+    if (this.nativeCss) {
+      let template = templateMap[is];
+      if (template && template.__applyShimInvalid) {
+        // update template
+        ApplyShim.transformRules(template._styleAst, is);
+        let target = this.nativeShadow ? template.content : null;
+        let placeholder = placeholderMap[is];
+        if (template._style) {
+          let s = template._style;
+          s.parentNode.removeChild(s);
+          template._style = null;
+        }
+        template._style = this._generateStaticStyle(host, template._styleAst, target, placeholder);
```

Comment:
We only need to update the textContent of the style in this case, isn't that so? Consider if it's worth the optimization to do that instead of replacing the style.